lib/shrine/attachment.rb

```diff
@@ -34,30 +34,28 @@ def initialize(name, **options)
       # Defines attachment methods for the specified attachment name. These
       # methods will be added to any model that includes this module.
       def define_attachment_methods!
-        attachment    = self
-        attacher_ivar = :"@#{@name}_attacher"
+        attachment = self
+        name = attachment_name
 
-        define_method :"#{@name}_attacher" do |options = {}|
-          if !instance_variable_get(attacher_ivar) || options.any?
-            instance_variable_set(attacher_ivar, attachment.build_attacher(self, options))
+        define_method "#{name}_attacher" do |reload = false, **options|
+          if reload || options.any? || !instance_variable_get("@#{name}_attacher")
+            instance_variable_set("@#{name}_attacher", attachment.build_attacher(self, options))
           else
-            instance_variable_get(attacher_ivar)
+            instance_variable_get("@#{name}_attacher")
           end
         end
 
-        module_eval <<-RUBY, __FILE__, __LINE__ + 1
-          def #{@name}=(value)
-            #{@name}_attacher.assign(value)
-          end
+        define_method "#{name}=" do |value|
+          send("#{name}_attacher").assign(value)
```

Comment:
It would be nice to convert these strings to symbols:

```rb
define_method :"#{name}=" do |value|
  send(:"#{name}_attacher").assign(value)
end
```

Reply:
Any reasons for this?

Reply:
No particular reason, sorry, feel free to disregard. It's just my preference, but I can add it after merging the PR.